pkg/checker/checker.go

```diff
@@ -0,0 +1,56 @@
+package checker
+
+import (
+	"fmt"
+	"net/http"
+	"time"
+
+	health "github.com/hellofresh/health-go"
+	"github.com/hellofresh/janus/pkg/api"
+	"github.com/hellofresh/janus/pkg/router"
+	log "github.com/sirupsen/logrus"
+)
+
+// Register registers the health checks for valid API definitions
+func Register(r router.Router, repo api.Repository) error {
+	definitions, err := repo.FindValidAPIHealthChecks()
+	if err != nil {
+		log.WithError(err).Error("Error fetching API definitions for health check registering")
+		return err
+	}
+
+	for _, definition := range definitions {
+		log.Debugf("%s health check registered", definition.Name)
+		health.Register(health.Config{
+			Name:      definition.Name,
+			Timeout:   time.Second * time.Duration(definition.HealthCheck.Timeout),
+			SkipOnErr: true,
+			Check:     check(definition),
+		})
+	}
+
+	r.GET("/status", health.HandlerFunc)
+	return nil
+}
+
+func check(definition *api.Definition) func() error {
+	return func() error {
+		req, err := http.NewRequest(http.MethodGet, definition.HealthCheck.URL, nil)
+		if err != nil {
+			log.WithError(err).Error("Creating the request for the health check failed")
+			return err
+		}
+
+		resp, err := http.DefaultClient.Do(req)
```

Comment:
Take a 👀 at [here](https://github.com/hellofresh/signin-service/blob/master/pkg/checker/auth_service.go) to avoid too many open files issue